src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -1010,24 +1010,28 @@ private void WindowClosed(object sender, EventArgs e)
         // passes it to thecurrent workspace
         void DynamoView_KeyDown(object sender, KeyEventArgs e)
         {
-            if (e.Key == Key.Escape)
-                dynamoViewModel.BackgroundPreviewViewModel.CanNavigateBackground = true;
+            if (e.Key == Key.Escape && this.IsMouseOver)
```

Comment:
An extra check is added here for `IsMouseOver`. We don't want to be able to override navigation when we're not over the Dynamo view.